### PR TITLE
DOC: Improved docs for `jnp.nanmin` and `jnp.nanmax`

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461

### CODE REVIEW

File: jax/_src/numpy/reductions.py [hunk 1]
```diff
@@ -1182,20 +1182,166 @@ def _nan_reduction(a: ArrayLike, name: str, jnp_reduction: Callable[..., Array],
...
 def nanmin(a: ArrayLike, axis: Axis = None, out: None = None,
            keepdims: bool = False, initial: ArrayLike | None = None,
            where: ArrayLike | None = None) -> Array:
+  r"""Return the minimum of the array elements along a given axis, ignoring NaNs.
+
+  JAX implementation of :func:`numpy.nanmin`.
+
+  Args:
+    a: Input array.
+    axis: int or array, default=None. Axis along which the minimum is computed.
```
Comment (dfm) [inline jax/_src/numpy/reductions.py:1196]:
```suggestion
    axis: int or sequence of ints, default=None. Axis along which the minimum is computed.
```

File: jax/_src/numpy/reductions.py [hunk 2]
```diff
@@ -1182,20 +1182,166 @@ def _nan_reduction(a: ArrayLike, name: str, jnp_reduction: Callable[..., Array],
...
+      If None, the minimum is computed along the flattened array.
+    keepdims: bool, default=False. If True, reduced axes are left in the result
+      with size 1.
+    initial: int or array, default=None. Initial value for the minimum.
+    where: int or array of boolean dtype, default=None. The elements to be used
```
Comment (dfm) [inline jax/_src/numpy/reductions.py:1201]:
```suggestion
    where: array of boolean dtype, default=None. The elements to be used
```

File: jax/_src/numpy/reductions.py [hunk 3]
```diff
@@ -1182,20 +1182,166 @@ def _nan_reduction(a: ArrayLike, name: str, jnp_reduction: Callable[..., Array],
...
 def nanmax(a: ArrayLike, axis: Axis = None, out: None = None,
            keepdims: bool = False, initial: ArrayLike | None = None,
            where: ArrayLike | None = None) -> Array:
+  r"""Return the maximum of the array elements along a given axis, ignoring NaNs.
+
+  JAX implementation of :func:`numpy.nanmax`.
+
+  Args:
+    a: Input array.
+    axis: int or array, default=None. Axis along which the maximum is computed.
```
Comment (dfm) [inline jax/_src/numpy/reductions.py:1278]:
```suggestion
    axis: int or sequence of ints, default=None. Axis along which the maximum is computed.
```

File: jax/_src/numpy/reductions.py [hunk 4]
```diff
@@ -1182,20 +1182,166 @@ def _nan_reduction(a: ArrayLike, name: str, jnp_reduction: Callable[..., Array],
...
+      If None, the maximum is computed along the flattened array.
+    keepdims: bool, default=False. If True, reduced axes are left in the result
+      with size 1.
+    initial: int or array, default=None. Initial value for the maximum.
+    where: int or array of boolean dtype, default=None. The elements to be used
```
Comment (dfm) [inline jax/_src/numpy/reductions.py:1283]:
```suggestion
    where: array of boolean dtype, default=None. The elements to be used
```

**Review by dfm — CHANGES_REQUESTED** [review]
Thanks! Tiny changes requested.

File: jax/_src/numpy/reductions.py (hunk 1 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 2 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 3 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 4 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

**Comment by rajasekharporeddy** [comment]
Modified the docstrings and squashed the commits into one.

**Review by dfm — APPROVED** [review]

**Pull request merged**